be/src/vec/common/hash_table/partitioned_hash_table.h

```diff
@@ -0,0 +1,666 @@
+            level0_sub_table.prefetch(key_holder);
+        }
+    }
+
+    template <bool READ>
+    void ALWAYS_INLINE prefetch_by_hash(size_t hash_value) {
+        if constexpr (ENABLE_PARTITIONED) {
+            if (_is_partitioned) {
+                const auto bucket = get_sub_table_from_hash(hash_value);
+                level1_sub_tables[bucket].template prefetch_by_hash<READ>(hash_value);
+            } else {
+                level0_sub_table.template prefetch_by_hash<READ>(hash_value);
+            }
+        } else {
+            level0_sub_table.template prefetch_by_hash<READ>(hash_value);
+        }
+    }
+
+    template <bool READ, typename KeyHolder>
+    void ALWAYS_INLINE prefetch(KeyHolder& key_holder) {
+        if constexpr (ENABLE_PARTITIONED) {
+            if (_is_partitioned) {
+                const auto& key = key_holder_get_key(key_holder);
+                const auto key_hash = hash(key);
+                const auto bucket = get_sub_table_from_hash(key_hash);
+                level1_sub_tables[bucket].template prefetch<READ>(key_holder);
+            } else {
+                level0_sub_table.template prefetch<READ>(key_holder);
+            }
+        } else {
+            level0_sub_table.template prefetch<READ>(key_holder);
+        }
+    }
+
+    /** Insert the key,
+      * return an iterator to a position that can be used for `placement new` of value,
+      * as well as the flag - whether a new key was inserted.
+      *
+      * You have to make `placement new` values if you inserted a new key,
+      * since when destroying a hash table, the destructor will be invoked for it!
+      *
+      * Example usage:
+      *
+      * Map::iterator it;
+      * bool inserted;
+      * map.emplace(key, it, inserted);
+      * if (inserted)
+      *     new(&it->second) Mapped(value);
+      */
+    template <typename KeyHolder>
+    void ALWAYS_INLINE emplace(KeyHolder&& key_holder, LookupResult& it, bool& inserted) {
+        size_t hash_value = hash(key_holder_get_key(key_holder));
+        emplace(key_holder, it, inserted, hash_value);
+    }
+
+    /// Same, but with a precalculated values of hash function.
+    template <typename KeyHolder>
+    void ALWAYS_INLINE emplace(KeyHolder&& key_holder, LookupResult& it, bool& inserted,
+                               size_t hash_value) {
+        if constexpr (ENABLE_PARTITIONED) {
+            if (_is_partitioned) {
+                size_t buck = get_sub_table_from_hash(hash_value);
+                level1_sub_tables[buck].emplace(key_holder, it, inserted, hash_value);
+            } else {
+                level0_sub_table.emplace(key_holder, it, inserted, hash_value);
+                if (UNLIKELY(level0_sub_table.need_partition())) {
+                    convert_to_partitioned();
+
+                    // The hash table was converted to partitioned, so we have to re-find the key.
+                    size_t buck = get_sub_table_from_hash(hash_value);
```

Comment:
In this case, the  it should be changed, in your code, it is still the it in old level0 hash table.